contrib/cirrus/check_image.sh

```diff
@@ -36,4 +36,28 @@ do
         "$(systemctl list-unit-files --no-legend $REQ_UNIT)" = "$REQ_UNIT enabled" || let "RET+=1"
 done
 
+# Previously, perdiodic package updates were enabled and would trigger
+# soon after an Ubuntu VM booted, grabbing a specific lock file for
+# several minutes after setup_environment.sh ran (previous script here also).
+DPKGLCK=/var/lib/dpkg/lock
+if [[ "$OS_RELEASE_ID" == "ubuntu" ]] && [[ -r "$DPKGLCK" ]]
+then
+    item_test "The lock file $DPKGLCK exists" -r "$DPKGLCK" || let "RET+=1"
+    flock --exclusive --nonblock --no-fork $DPKGLCK sleep 1h &
+    sleep 1s  # just in case
+    FLOCKPID="$!"
+    item_test "Uncontested dpkg lock acquired" "$(jobs -r | wc -l)" -eq "1" || let "RET+=1"
+    echo "(Waiting 60 seconds for contestents to block)"
+    sleep 60s
+    # exit(1) means no processes found
+    CONTESTED=$(lsof -p ^$FLOCKPID -t -f -- $DPKGLCK &> /dev/null; echo $?)
+    item_test "No processes blocked locked $DPKGLCK w/in 1 minute" "$CONTESTED" -eq "1" || let "RET+=1"
+    item_test "Test locking process is killed (ignore terminated message)" $(kill $FLOCKPID; echo $?) -eq "0"
+elif [[ "$OS_RELEASE_ID" == "ubuntu" ]]
+then
+    echo "not ok Ubuntu has a $DPKGLCK lock file"
+    let "RET+=1"
+fi
+
+echo "Total failed tests: $RET"
```

Comment:
The above is complicated, I don't like it very much.  At the same time, it would catch the problem that is fixed by the `systemd_banish()` (on Ubuntu) :frowning_face: